opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/drools/DroolsAlarmContext.java

```diff
@@ -319,7 +323,21 @@ public void handleNewOrUpdatedAlarm(OnmsAlarm alarm) {
                     .getId())
                     .ifPresent(acks::add);
         } else {
-            acks.addAll(acknowledgmentDao.findLatestAcks());
+            // Calculate the creation time of the earliest alarm
+            final Date earliestAlarm  = alarms.stream()
+                    .map(OnmsAlarm::getFirstEventTime)
+                    .filter(Objects::nonNull)
+                    .min(Comparator.naturalOrder())
+                    .orElseGet(() -> {
+                        // We didn't find any dates - either the set is empty (in which case this function
+                        // wouldn't be called) or all the dates are null (which they shouldn't be.)
+                        // Let's log an error, and return some date for sanity
+                        final LocalDateTime oneMonthAgoLdt = LocalDateTime.now().minusMonths(1);
+                        final Date oneMonthAgo = Date.from(oneMonthAgoLdt.atZone(ZoneId.systemDefault()).toInstant());
+                        LOG.error("Could not find minimum alarm creation time for alarms: {}. Using: {}", oneMonthAgo);
```

Comment:
Looks like this is missing the alarms as a parameter.

Reply:
Good catch.